backends/dialogs/gtk/gtk-dialogs.cpp

```diff
@@ -35,25 +35,20 @@
 
 #include <gtk/gtk.h>
 
-Common::DialogManager::DialogResult GtkDialogManager::showFileBrowser(const char *title, Common::FSNode &choice, bool isDirBrowser) {
+Common::DialogManager::DialogResult GtkDialogManager::showFileBrowser(const Common::U32String &title, Common::FSNode &choice, bool isDirBrowser) {
 	if (!gtk_init_check(NULL, NULL))
 		return kDialogError;
 
 	DialogResult result = kDialogCancel;
 
-	// Get current encoding
-	Common::String guiEncoding = "ASCII";
-#ifdef USE_TRANSLATION
-	guiEncoding = TransMan.getCurrentCharset();
-#endif
-	Common::Encoding utf8("utf-8", guiEncoding);
-
 	// Convert labels to UTF-8
-	char *utf8Title = utf8.convert(title, strlen(title));
-	Common::String choose = _("Choose");
-	char *utf8Choose = utf8.convert(choose.c_str(), choose.size());
-	Common::String cancel = _("Cancel");
-	char* utf8Cancel = utf8.convert(cancel.c_str(), cancel.size());
+	Common::String titleStr = title.encode();
+	Common::String choose = _("Choose").encode();
+	Common::String cancel = _("Cancel").encode();
+
+	char *utf8Title = (char *)(titleStr.c_str());
+	char *utf8Choose = (char *)(choose.c_str());
+	char *utf8Cancel = (char *)(cancel.c_str());
```

Comment:
GTK documentation for dialogs (https://developer.gnome.org/gtk3/stable/GtkFileChooserDialog.html) shows that the functions that use these titles are of type `const gchar*`, and `gchar` is just a typedef for `char`. So, I'm pretty sure we could further remove these lines, and below (where we actually use strings) we can pass `str.c_str()` and everything should work fine. Just wanted to get a second opinion before I push this out though.

Reply:
When I saw those lines I assumed that the `gtk_file_chooser_dialog_new()` function needed `char *` pointers, while `String::c_str()` returns a `const char *`. But you are right and indeed it takes a `const gchar *`. So I have no idea why those casts to `char *` are here. Maybe the `gtk_file_chooser_dialog_new()` was changed and in the past it was indeed taking non-const `gchar *` strings? But even in the gtk2 documentation (https://developer.gnome.org/gtk2/stable/GtkFileChooserDialog.html) it takes const ones.
So I think you are right and this code can be simplified further.

Reply:
Thanks for the response. I have changed it.